app/models/common_tagging.rb

```diff
@@ -8,22 +8,30 @@
 class CommonTagging < ApplicationRecord
   # we need "touch" here so that when a common tagging changes, the tag(s) themselves are updated and
   # they get noticed by the tag sweeper (which then updates their autocomplete data)
-  belongs_to :common_tag, class_name: 'Tag', touch: true
+  belongs_to :common_tag, class_name: "Tag", touch: true
   belongs_to :filterable, polymorphic: true, touch: true
 
-  validates_presence_of :common_tag, :filterable, message: "does not exist."
-  validates_uniqueness_of :common_tag_id, scope: :filterable_id
+  validates :common_tag, :filterable, presence: { message: "does not exist." }
+  validates :common_tag_id, uniqueness: { scope: :filterable_id }
 
   after_create :update_wrangler
   after_create :inherit_parents
   after_create :remove_uncategorized_media
 
   after_commit :update_search
 
+  before_destroy :check_unwrangleable
+  # Check an unwrangleable tag has at least 1 fandom before destroying the association
+  def check_unwrangleable
+    tag = Tag.find_by(id: common_tag)
+    return unless tag&.unwrangleable && tag.parents.by_type("Fandom").size == 1
```

Comment:
Ah, are you talking about cascading deletes (I'm assuming when a tag is deleted all associations go with it)? I hadn't thought of that at all.  If we have an unwrangleable relationship (in a single fandom), and then one of the characters gets deleted, will the relationship end up not deleted because of this?

Reply:
I wasn't talking about cascading deletes, sorry -- I meant something like this case:
```
f = create(:canonical_fandom)
c = create(:canonical_character)
r = create(:relationship)
rf = create(:common_tagging, common_tag: r, filterable: f)
rc = create(:common_tagging, common_tag: r, filterable: c)
r.update(unwrangleable: true)
expect(rc.destroy).to be_truthy
```
Not removing a `Character`, but removing the `CommonTagging` linking a `Character` and a `Relationship`. If the `Relationship` is unwrangleable, and has exactly one `Fandom` parent, then to this code it'll look like the fandom is being removed from an unwrangleable relationship (even though the fandom is staying right where it is).

(Though now that you mention cascading deletes, it occurs to me that it might be good to check whether an unwrangleable tag can be destroyed properly with this code. It might be necessary to check `destroyed_by_association` here.)